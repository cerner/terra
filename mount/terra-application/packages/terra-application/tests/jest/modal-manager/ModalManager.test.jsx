import React from 'react';
import ModalManager from '../../../src/modal-manager';

const TestChild = () => <div />;

describe('ModalManager', () => {
  describe('Snapshots', () => {
    it('should render with required props', () => {
      const wrapper = shallow((
        <ModalManager>
          <TestChild />
        </ModalManager>
      ));

      expect(wrapper).toMatchSnapshot();
    });

    it('should render with optional props', () => {
      const wrapper = shallow((
        <ModalManager
          disclosureAccessory={<div>Test Accessory</div>}
          navigationPromptResolutionOptions={{
            test: 'options',
          }}
        >
          <TestChild />
        </ModalManager>
      ));

      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('DisclosureContainer', () => {
    it('should wrap disclosed content in a DisclosureContainer with default prompt options', () => {
      const promptOptions = { test: 'options' };

      const wrapper = shallow((
        <ModalManager
          navigationPromptResolutionOptions={promptOptions}
        >
          <TestChild />
        </ModalManager>
      ));

      const disclosureWrapper = wrapper.props().withDisclosureContainer(<div>Test Disclosure Content</div>);
      expect(disclosureWrapper.props.navigationPromptResolutionOptions).toBeDefined();
    });
  });
});
