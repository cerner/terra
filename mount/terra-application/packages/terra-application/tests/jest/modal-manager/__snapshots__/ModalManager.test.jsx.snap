// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalManager Snapshots should render with optional props 1`] = `
<ModalManager
  disclosureAccessory={
    <div>
      Test Accessory
    </div>
  }
  withDisclosureContainer={[Function]}
>
  <TestChild />
</ModalManager>
`;

exports[`ModalManager Snapshots should render with required props 1`] = `
<ModalManager
  withDisclosureContainer={[Function]}
>
  <TestChild />
</ModalManager>
`;
